Components/Queues/Handler/PhpAmqpHandler.php

```diff
@@ -0,0 +1,230 @@
+            $this->stop();
+            if ($this->channel && $this->channel->getConnection()) {
+                $this->channel->basic_cancel($consumerTag, true, true);
+            }
+        } catch (\Exception $exception) {
+            $this->log($exception->getMessage());
+            exit(1);
+        }
+    }
+
+    /**
+     * Verifies if the message object is an instance of QueueMesseageInterface
+     * and if the handler is not an instance of QueueMessageHandlerInterface. Used by dispatchMessage function
+     * @param \Smartbox\Integration\FrameworkBundle\Core\Messages\MessageInterface $message
+     * @return bool
+     */
+    private function isQueueMessage($message): bool
+    {
+        return $message instanceof QueueMessageInterface && !($this->endpoint->getHandler() instanceof QueueMessageHandlerInterface);
+    }
+
+    /**
+     * Log in console the message passed as param
+     * @param string $message
+     * @param array $ctx
+     * @throws \Exception
+     */
+    private function log(string $message, array $ctx = [])
+    {
+        if (null === $this->logger) {
+            return;
+        }
+
+        $now = new \DateTime();
+        $ctx['time'] = $now->format('Y-m-d H:i:s.u');
+        $this->logger->info($message, $ctx);
+    }
+
+    /**
+     * Dispatch the message to target system or events to database
+     * @param QueueMessage $message
+     * @throws \Exception
+     */
+    public function dispatchMessage(QueueMessage $message): void
+    {
+        if (!$this->endpoint) {
+            $this->getExceptionHandler()('Endpoint is undefined');
+            return;
+        }
+
+        try {
+            if ($this->isQueueMessage($message) && null !== $this->smartesbHelper) {
+                $endpoint = $this->smartesbHelper->getEndpointFactory()->createEndpoint($message->getDestinationURI(), EndpointFactory::MODE_CONSUME);
+                $endpoint->getHandler()->handle($message->getBody(), $endpoint);
+            } else {
+                $this->endpoint->handle($message);
+            }
+        } catch (\Exception $exception) {
+            $this->getExceptionHandler()($exception, ['headers' => $message->getHeaders(), 'body' => $message->getBody()]);
+        }
+    }
+
+    /**
+     * @param AMQPMessage
+     * @return  \Smartbox\Integration\FrameworkBundle\Core\Messages\MessageInterface|QueueMessageInterface $message
+     */
+    public function deserializeMessage(AMQPMessage $message): QueueMessageInterface
+    {
+        try {
+            return $this->serializer->deserialize($message->getBody(), SerializableInterface::class, $this->format);
+        } catch (\Exception $exception) {
+            $this->getExceptionHandler()($exception, ['headers' => $message->getHeaders(), 'body' => $message->getBody()]);
+        }
+    }
+
+    /**
+     * Set a AMQPChannel object to class variable
+     * @param AMQPChannel $channel
+     */
+    public function setChannel(AMQPChannel $channel): void
```

Comment:
The feature of marking the methods as return void was introduced in php 7.1 and framework bundle needs to support php 7.0

Reply:
All the return with type void was removed from the code.